src/common/film.c

```diff
@@ -90,7 +90,11 @@ int32_t dt_film_get_id(const char *folder)
   int32_t filmroll_id = -1;
   sqlite3_stmt *stmt;
   DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+#ifdef _WIN32
+                              "SELECT id FROM main.film_rolls WHERE folder LIKE ?1",
```

Comment:
LIKE is case sensitive, so why it is needed as it seems equivalent to the "=" just below, no?

Reply:
Oh, no LIKE is no case  sensitive. To avoid LIKE (maybe slower) we also could use:

`... WHERE folder = ?1 COLLATE NOCASE`

Reply:
But then, we may also some similar fixes for `filename`? All in all I fear this is opening a can of worms :(

Reply:
I don't know what is faster. If there is a place where performance is critical it's in collection for example (a lot of LIKEs there) but film_roll recognition is not as critical (in import_session variables' pathname is even evaluated 3 times...).

> But then, we may also some similar fixes for filename?

I think filename works as it is because of the mechanism which tests if the file already exists and increments the index if necessary (here linix and windows answer differently but consistently).
With film_roll on the contrary we have to reuse it if already exists.

> All in all I fear this is opening a can of worms :(

Agreed, especially on collect side. See #11288 for example. But if we can afford some issues in collect, import is more critical as it can feed inconsistencies into the db.